components/brave_news/browser/feed_generation_info.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_news/browser/feed_generation_info.h"
+
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/contains.h"
+#include "base/containers/span.h"
+#include "base/ranges/algorithm.h"
+#include "brave/components/brave_news/browser/feed_sampling.h"
+#include "brave/components/brave_news/browser/publishers_controller.h"
+#include "brave/components/brave_news/browser/signal_calculator.h"
+#include "brave/components/brave_news/common/brave_news.mojom.h"
+
+namespace brave_news {
+
+FeedGenerationInfo::FeedGenerationInfo(
+    const std::string& locale,
+    const FeedItems& feed_items,
+    const Publishers& publishers,
+    std::vector<std::string> channels,
+    const Signals& signals,
+    const std::vector<std::string>& suggested_publisher_ids,
+    const TopicsResult& topics)
+    : locale_(locale),
+      channels_(std::move(channels)),
+      suggested_publisher_ids_(suggested_publisher_ids),
```

Comment:
Need to have `suggested_publisher_ids_`?

Reply:
Yup, we need a copy of the suggested_publisher_ids to back the span (which we change instead of removing items from the vector).